### ISSUE TITLE
'Stick' damage

### ISSUE THREAD

**Issue by XenonLog01** [Anuken/Mindustry-Suggestions#2304]
**Describe the content or mechanics you are proposing.**
When turrets firing metaglass shoot, there is a chance it will fracture and 'stick' to the target for a time, dealing damage at a regular interval. This would be similar to burning, but could be removed by segments **and** water turrets.

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**
It will make for more interesting combat, and might save people's bases due to damage being slowly inflicted. It will increase the use of metaglass in turrets, and by extent the use of turrets that can shoot it. 

**Implementation**:
Just an Idea, but turrets could have 'velocity', which would determine the likelihood of fragments sticking, and the length of time they are stuck, as well as the amount of damage inflicted per x time. The likelihood of a turret firing water removing a bit of shrapnel could be determined again, by it's velocity. Velocity could be increased with power or liquids, making them even better. To cap it off, velocity could have an effect on damage, determining a modifier to apply to the damage. 

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [X] I have done a quick search in the list of suggestions to make sure this has not been suggested yet. <I searched, but this might already be suggested, and it has a vastly different title. If that's the case, sorry.
2. - [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
3. - [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
4. - [X] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by XenonLog01** [comment]
Again, if this was suggested, sorry.

**Comment by octopuscaat** [comment]
yes, but, does it really _fit_ with the game

**Comment by XenonLog01** [comment]
Honestly, it seems to fit, but IDK, I don't do the coding or game design, so It's not up to me, but to the devs. I'll ask #development about it.

**Comment by Awesomespore** [comment]
It makes sense until you consider the fact that robots don't have flesh, then it kinda starts to fall apart.
Not a bad idea though, it just doesn't make a lot of sense when you think about it.

**Comment by Kyllingene** [comment]
Well, you could always say that since metaglass is made with lead, and thus metalized, that the metaglass is short-circuiting the unit on a small scale.

**Comment by notrealn** [comment]
glass isnt conductive, theres no way it could short circut units

copper is conductive, why isnt that being suggested

anyways I dont think this fits in with mindustry, robots aren't affected by shrapnel in them

**Comment by Kyllingene** [comment]
Reading is very fun!

> since metaglass is made with lead, and thus metalized,

See it now?

> since metaglass is made with ***lead***, and thus ***metalized***,

**Comment by notrealn** [comment]
pretty sure mixing dirt with metals doesnt suddently make it conductive, but im not an expert

**Comment by Kyllingene** [comment]
Enough to make it conductive. Also, you phrased it poorly; if you mixed equal parts lead and sand, with a particular process, a particular enough process to make the glass transparent, you would get a conductive result. And even if that wasn't quite true, it's true enough to make the (already invalid) argument of "but realism" go away. We have drills and conveyors that work for no power.

**Comment by notrealn** [comment]
not only does this not seem very realistic, it just seems like unnecesarry features that make the game more complex for no reason
why does everything need to be given a DOT?

**Comment by Kyllingene** [comment]
Ah, an actual argument! Alas, this is a poor one; VERY little added complexity (if you can't understand that metaglass deals damage over time, you can't understand that it frags), and with very little programming effort. If this is unnecessary, let's remove slags damage over time too! While we're at it, why did we ever get power diodes? They have mostly fringe uses, and are almost never necessary. But they took a fair bit of programming to add, and require much more effort to understand.

**Comment by notrealn** [comment]
It doesnt make much sense to give something with already frag abilities also the ability to do damage over time. Slag's whole gimmick is that its hot and melts units over time, but glass is just a bullet thats supposed to do damage. The name of this DOT isnt very good either, its literally called "stick" damage (not very descriptive).
Anyways, while were at it, why not make everything do DOT? it doesnt add much complexity right??? and who cares if it doesnt make much sense?

**Comment by Kyllingene** [comment]
The sticking part of your argument: *metaglass* ***frags***, *and thus is perfect for damage over time.* No other materials apply themselves as readily, and your argument of making everything deal damage over time makes no sense because metaglass, again, frags, unlike other materials, and we only need one damage-over-time bullet.

> ...whole gimmick... but glass is just a bullet...

Did you stop to realize that, if this is implemented, metaglass' gimmick *will be* the damage over time?

**Comment by notrealn** [comment]
> metaglass frags, and thus is perfect for damage over time
No? Every ammo in this game only has 1 special property (except swarmer surge, which has homing and lightning and I guess liquids in wave/tsunami cause they also do knockback), why should glass be given 2?
> because metaglass, again, frags, unlike other materials
yeah except for plast but whatever, which is also mostly metal and therefore conductive which somehow makes it short circut robots just by sticking them with it?
> we only need one damage-over-time bullet.
we dont neeeeeeed a damage over time bullet. this already exists in pyratite, coal, and slag. If you want damage over time, you should have to get special ammo thats just for it, not something ordinary like glass

**Comment by Kyllingene** [comment]
> yeah except for plast but whatever,

Then fine, make plastanium deal damage over time.

> this already exists in pyratite, coal, and slag.

Slag is liquid-only, and very few turrets can take both pyratite / coal and metaglass.

**Comment by notrealn** [comment]
> very few turrets

you mean 0? why cant damage over time be limited to only a few special turrets anyways?

anyways, this suggestion is very vague on how strong this should be and how it should be implemented. this extra "velocity" thing just seems like a bunch of extra work just for a bit of extra DOT. Another thing is why would segments remove this "sticking" effect from turrets? segments aren't for shooting units, they are for shooting bullets in the air. all of this just seems like so much work for a change that isnt very interesting and doesnt make much sense.

**Comment by Kyllingene** [comment]
> why cant damage over time be limited to only a few special turrets anyways?

Not many turrets use metaglass anyway.

> anyways, this suggestion is very vague on how strong this should be and how it should be implemented.

How should it be implemented besides with a status effect, which accomplishes it perfectly and with little to no actual programming effort, or even knowledge, required?

And as to not indicating strength, why is that such a problem? Many developers hate overly specific suggestions.

> this extra "velocity" thing just seems like a bunch of extra work

Actually, looking at the source code, it would be really quite easy to do that. It's just another stat in the bullet type constructor. Also, think about the modding potential; not only fragmenting, damage-over-time bullets, but a velocity stat to play with!

>  segments aren't for shooting units,

I agree, the segment idea is a little odd.

**Comment by notrealn** [comment]
> Not many turrets use metaglass anyway.

3 turrets use frag bullets. btw 10 turrets have the ability to do damage over time (more than half!!!), we really need more?

> but a velocity stat to play with!

bullets already have a velocity............... how else would they move?

this still seems to me like some random suggestion that doesnt add much depth to the game.

**Comment by Kyllingene** [comment]
> bullets already have a velocity............... how else would they move?

*Turret* velocity, as in a stat of the turret, independent of ammo.

> 3 turrets use frag bullets

**Comment by notrealn** [comment]
> Turret velocity, as in a stat of the turret, independent of ammo.

umm what?? turret velocity? how does that make any sense? bullets already have a speed, why would turrets need a velocity? also, isnt this for frag bullets which lose all speed when it explodes and spreads out? how does a turret's velocity have anything to do with the bullet?

**Comment by Kyllingene** [comment]
> umm what?? turret velocity? how does that make any sense?

You see, turrets are in fact the thing that gives bullets their velocity.

> bullets already have a speed,

Look at mods. Turrets can either use built-in bullets, or make their own *turret-to-turret*.

> also, isnt this for frag bullets which lose all speed when it explodes and spreads out? 

You see, as metaglass is not explosive, it is the force of impact which explodes the bullet. So, the faster it flies (and thus hits), the faster the fragments fly, and thus the more likely a fragment is to stick. Please consult physics before you quote it.

**Comment by notrealn** [comment]
yeah, except frag bullets dont follow any physics. If glass bullets shatter on impact, why do their fragments eveny spread out in all directions instead of inheriting the speed they were origionally going at? physics dont apply to most things in this game anyways.

> Look at mods. Turrets can either use built-in bullets, or make their own turret-to-turret.

so does this mean this should only apply to modded bullets? why are you bringing up mods here? almost all turrets included in the base game use built in ones or one created inline with a constructor (like the one below), correct? so in conclusion **how does a turret's velocity affect how fast it's bullet will travel**, and how does that apply to frag bullets that don't inherit speed from its original projectile?

![image](https://user-images.githubusercontent.com/56411504/115120502-d51a5a80-9f7b-11eb-95b8-167a48ddbb84.png)

**Comment by notrealn** [comment]
Another thing is that computers in this game dont even short-circut, so your "logical" explanation as to how this would even make sense doesnt seem very legit. Covering a unit in water doesnt make it short circut and zapping a unit with lightning should just fry every single component inside it instead of doing some damage, right?

so basically it doesnt make sense, doesnt add much to the game, and isnt that great of a suggestion.

**Comment by Kyllingene** [comment]
> spread out in all directions instead of inheriting the speed they were origionally going at?

Do you expect the fragments to all go into the wall / unit that gets hit? Try throwing an egg at a wall. Do you expect every bit of the egg to hit the wall, or do you expect a lot of it to fly a little way away from the wall? Now try throwing a glass. Now try throwing a large reinforced glass pellet designed to fracture out of a cannon.

> included in the base game use built in ones or one created inline with a constructor (like the one below), correct?

Correct. That was exactly my point. No, it does not apply only to mods.

> so in conclusion how does a turret's velocity affect how fast it's bullet will travel,

Well, when a turret makes a bullet appear by shooting it, it uses a function to make it exist. There are parameters which must be passed. How hard would it be to simply add a velocity parameter, and fill that parameter with the turrets velocity stat? Not at all. And, in physics terms, I shouldn't have to explain that when you throw an egg, *you* determine its speed, not the egg. I will throw an egg at a different speed than you will.

> Another thing is that computers in this game dont even short-circut

Excuse me, what? What proof do you have of that? Why do you assume that these computers, which use electrical components, are immune to conductive materials bridging two previously unconnected paths? That is a horrendous leap to make.

> Covering a unit in water doesnt make it short circut and zapping a unit with lightning should just fry every single component inside it instead of doing some damage, right?

You see, there are these super ridiculously high-tech things from 3021 called *water-proof phones* (sarcasm alert, just in case). We apply the same classified, alien tech to our units. And of course, zapping it makes sense - until you realize that when you zap you *do* deal damage, and you would be hard pressed to make solid lightning, thus the damage is mostly short-circuiting. The units are not absolutely immune, but aren't immediately destroyed by it (except for weaker, less advanced units). If short-circuiting should instantly kill a unit, why didn't you mention that the shrapnel should instantly kill the units?

so basically your points dont make sense, dont have good bases (neither in terms of physics nor in terms of gameplay), and arent that well thought-out.

**Comment by notrealn** [comment]
> so basically your points dont make sense, dont have good bases (neither in terms of physics nor in terms of gameplay), and arent that well thought-out.

same to you man. if you short circut a machine, it doesnt work anymore. everything is fried, your computer wont continue working at all even after you dry it. mindustry is a videogame, its not supposed to be super realistic. after you "short circut" a dagger with lightening, it still moves and shoots the same way it did when it was at full hp.

> you determine its speed, not the egg

but thats not the case in this game. when creating bullets with constructors, they will have the speed you give them. making another property for turrets for bullet speed is redundant. 

> sarcasm alert, just in case

do you really have to treat everyone on the internet like 3 year olds? give me some credit at least, man. you dont have to act like natalie in every way

>  Do you expect every bit of the egg to hit the wall, or do you expect a lot of it to fly a little way away from the wall?

What i dont expect is for the egg's fragments to evenly spread out in a circular pattern, and then having those shards suddently expire once they reach an arbitrary number of seconds. 
Anyways, even when glass bullets dont hit anything they still explode (so they are explosive right? somehow graphite does aoe damage when shot by a hail/ripple), and they definetly arent hitting the floor as they can still hit air units and fly straight over walls.

**Comment by Kyllingene** [comment]
> mindustry is a videogame, its not supposed to be super realistic

*Then why do you keep attacking this suggestion on the grounds of realism?*

> but thats not the case in this game. when creating bullets with constructors, they will have the speed you give them.

You literally just reiterated what I said. 

> making another property for turrets for bullet speed is redundant.

Turret velocity would affect not only bullet speed (thus giving even more customization in mods, without compromising the base game), but other statistics of a bullet, such as fragmentation, and possibly even penetration.

> What i dont expect is for the egg's fragments to evenly spread out in a circular pattern, and then having those shards suddently expire once they reach an arbitrary number of seconds.

They do not spread out evenly, actually. They go out in random directions. Look at the source code. And then they fall due to this little thing called *gravity*. As to them disappearing while inside a unit, well, they just *fell*. And all of these details are unnecessary, because as you pointed out yourself,

> mindustry is a videogame, its not supposed to be super realistic

**Comment by notrealn** [comment]
> Then why do you keep attacking this suggestion on the grounds of realism?

it doesnt make any sense and doesnt have any video game logic behind it. electricity is supposed to do damage in videogames, not OHKO robots. frag bullets are just supposed to just do damage, not also have some sort of DOT with "sticking". as i said before, its just an unnececary feature that isnt present in any other game, so nobody will expect it, and doesnt make any sense.

> Turret velocity would affect not only bullet speed (thus giving even more customization in mods, without compromising the base game), but other statistics of a bullet, such as fragmentation, and possibly even penetration.

so you are saying we should add a "velocity" stat that doesnt actually change the speed of bullets. good stuff

> And then they fall due to this little thing called gravity.

I literally explained that its impossible that gravity could be affecting these bullets. why can they still just fly over copper walls after falling?

**Comment by Kyllingene** [comment]
> so you are saying we should add a "velocity" stat that doesnt actually change the speed of bullets. good stuff

***Not only affects bullet speed.*** As in, it *does* affect bullet speed, but also affects other stuff.

> it doesnt make any sense and doesnt have any video game logic behind it. electricity is supposed to do damage in videogames, not OHKO robots

***That is exactly what I am proposing.***

> its just an unnececary feature that isnt present in any other game, so nobody will expect it, and doesnt make any sense.

Actually, I have seen several games with extremely similar features. Why do you think this is rare?

> I literally explained that its impossible that gravity could be affecting these bullets. why can they still just fly over copper walls after falling?

Well then, complain to Anuke about how bullets work then. It has nothing to do with frag bullets.

**Comment by notrealn** [comment]
> Not only affects bullet speed. As in, it does affect bullet speed, but also affects other stuff.

but why cant we just use bullet speed instead of this "turret velocity" then? this is just an extra layer of redundancy, turrets already can have unique bullets with speeds so why does the turret need to be given this?

> That is exactly what I am proposing.

you want electicity damage to ohko robots???? 

> Actually, I have seen several games with extremely similar features. Why do you think this is rare?

yes, usually in the form of bleed, but these are robots. they dont bleed, they melt/burn which already happens. there is no reason to add another "sticking" DOT to literal pieces of glass. if you want DOT, just use meltdown or something.

> Well then, complain to Anuke about how bullets work then. It has nothing to do with frag bullets.

maybe its cause that would make the bullets inconsistent? what if the true reason why frag bullets spread out like that is because this is a videogame, and turret ammo can be given properties like explosions (like silicon in ripples)? if bullets "fell", then anti air weapons would be useless against flying units long range. 

and anyways, if we both agree that "its a videogame" and "it doesnt have to be realistic", why are both of us trying to give a logical explanation on why it would/wouldnt happen?

**Comment by Kyllingene** [comment]
> but why cant we just use bullet speed instead of this "turret velocity" then? this is just an extra layer of redundancy, turrets already can have unique bullets with speeds so why does the turret need to be given this?

Because most turrets use the built-ins.

> if you want DOT, just use meltdown or something.

Ah yes, because the turret with the highest damage in the game is perfect for damage over time.

> maybe its cause that would make the bullets inconsistent? what if the true reason why frag bullets spread out like that is because this is a videogame

Yeah. ***It isn't always perfectly realistic.***

> and anyways, if we both agree that "its a videogame" and "it doesnt have to be realistic", why are both of us trying to give a logical explanation on why it would/wouldnt happen?

Because you keep attacking this suggestion based on realism. It doesn't matter whether or not it is realistic, but since it makes sense anyway, it can only strengthen my case.

**Comment by Awesomespore** [comment]
> Well, you could always say that since metaglass is made with lead, and thus metalized, that the metaglass is short-circuiting the unit on a small scale.

well I suppose yeah, you could say that it's getting lodged in loose wiring, etc.

**Comment by notrealn** [comment]
> Ah yes, because the turret with the highest damage in the game is perfect for damage over time.

or use pyratite swarmer, specter, salvo, slag tsunami, etc.

> It doesn't matter whether or not it is realistic, but since it makes sense anyway, it can only strengthen my case.

i agree, it doesnt really matter wether or not a video game mechanic is super realistic, but it does not make sense. I have never seen in any media that sticking any conductive material into a robot will make it take damage over time.

> well I suppose yeah, you could say that it's getting lodged in loose wiring, etc.

still doesnt explain how units would short circut and why they dont just stop working after short circuting.

**Comment by Kyllingene** [comment]
Then we have differing opinions on that. However, I believe that either way, it is close enough for even those focused on realism could accept it.

> still doesnt explain how units would short circut and why they dont just stop working after short circuting.

There are ways to mitigate damage from short circuits (for example, diodes). However, no defenses are perfect (for example, diodes stop working at a certain reverse voltage).

**Comment by notrealn** [comment]
lemme rephrase that: it doesnt explain how short circuting a unit will do damage over time

wouldn't it make more sense that short circuting a machine will have an almost instant effect, not one that takes multiple seconds?

**Comment by Kyllingene** [comment]
Prolonged voltage in the wrong way could most certainly cause problems over time. Modern technology already has similar flaws.
